src/java.desktop/windows/classes/com/sun/java/swing/plaf/windows/WindowsMenuItemUI.java

```diff
@@ -201,8 +202,9 @@ static void paintMenuItem(WindowsMenuItemUIAccessor accessor, Graphics g,
 
         if (lh.getCheckIcon() != null && lh.useCheckAndArrow()) {
             Rectangle rect = lr.getTextRect();
-
-            rect.x += lh.getAfterCheckIconGap();
+            if (menuItem.getHorizontalTextPosition() != SwingConstants.LEADING) {
```

Comment:
Not sure i understand why we only checking for "LEADING" text position. What if it is specified specifically as "LEFT" or "RIGHT"? What would result look like in the different component orientations with this fix?

Reply:
I just tested the latest version of the fix with the slightly modified test case (i added two extra menu items with 
`menuItem.setHorizontalTextPosition(SwingConstants.LEFT);`
and
`menuItem.setHorizontalTextPosition(SwingConstants.RIGHT);`
and it does not look exactly correct. Here's the screenshot:
<img width="331" height="511" alt="image" src="https://github.com/user-attachments/assets/8494241e-aefd-4259-bcb2-8b3cf2360d7d" />

Reply:
Fixed..it now looks ok to me...Please check

<img width="391" height="614" alt="image" src="https://github.com/user-attachments/assets/46661a18-947b-4483-969d-c7d05ace3ff0" />

Reply:
Yes, new version works correctly. I would suggest to add these menu items to the regression test but that is optional.